### DISCUSSION TITLE
FOSS License

### DISCUSSION THREAD

**Discussion by julianxhokaxhiu** [Vector35/binaryninja-api#1804] [General] [answered]
Hi,

my name is Julian and I'm one of the authors of https://github.com/julianxhokaxhiu/FFNx, an open-source driver for FF7 and FF8 which will allow you to re-play the games enhanced like never done before by Square-Enix.

The issue with this project although is that it requires ton of Reverse Engineering. I did try to use Cutter, Ghidra and other free tools around ( like Snowman ) but unfortunately all of them lack support for most of the C/C++ High level decompilation.

Your tool on the other side compares honestly with IDA Pro, which unfortunately is super expensive for a FOSS project. I did try your demo build and actually it decompiles ( also in HLIL ) the game engines quite fine, although the issue with the Demo is that it is too short in time, you can't save your progress and it's super slow to decompile under one single core.

Is there any plan on your side to provide a license which is "free only for FOSS projects"?
I'd be fine having that locked per PC, per Name, per Email or whatever restriction you may think of. Basically it could be a Personal license binded to a single person only.

I would be extremely happy if you could consider this option.

Thank you for now,
Julian

**Comment by psifertex** [comment, marked as answer]
This is something we've had requested a few times and we've debated different options. Our current thinking is that we're unlikely to offer a totally free license, but we might offer the student discount to developers of open source plugins under some restrictions. Sorry I don't have a better option for you right now, it's definitely something we've considered, just no decision at this time.

**Reply by julianxhokaxhiu** [reply]
Thank you for the reply!

Is there any plan in considering something like 99$/year price tag, like Apple does for it's Developer Account?

That could already be a good starter :)

Thank you in advance,
Julian

**Reply by psifertex** [reply]
Apple developer accounts don't include any of their actual licensed software so it's not a very good analogy. 😉 That's just paying for the privilege of putting your stuff in their store!

That said, if we did implement the educational pricing for OSS it would be as described here: https://faq.binary.ninja/en/do-you-have-a-student-discount-pF28YngkuxuuHWKTk

**Reply by julianxhokaxhiu** [reply]
Ah that works even better! Allright :) I'll stay tuned in case you ever launch such a program! Thank you
